Tests/SonarScanner.MSBuild.Tasks.UnitTest/GetAnalyzerSettingsTests.cs

```diff
@@ -283,20 +283,33 @@ public void ConfigExists_SettingsMerged()
 
         [DataTestMethod]
         [DataRow("7.3", "cs", DisplayName = "Legacy CS")]
-        [DataRow("7.4", "cs")]
         [DataRow("7.3", "vbnet", DisplayName = "Legacy VB")]
-        [DataRow("7.4", "vbnet")]
-        public void ConfigExists_ForProductProject_SonarAnalyzerSettingsUsed(string sonarQubeVersion, string language)
+        public void ConfigExists_ForLegacyProductProject_SonarAnalyzersAndConfigurationUsed(string sonarQubeVersion, string language)
         {
             // Arrange and Act
             var executedTask = Execute_ConfigExists(sonarQubeVersion, language, false, null);
 
             // Assert
             executedTask.RuleSetFilePath.Should().Be($@"c:\{language}-normal.ruleset");
+            // There are two Google.Protobuf.dll as one is from the C# analyzer and one from the VBNet analyzer.
```

Comment:
The legacy code does only a [selectMany](https://github.com/SonarSource/sonar-scanner-msbuild/blob/05499245d82522b5543db07355cc20b10631335a/src/SonarScanner.MSBuild.Tasks/GetAnalyzerSettings.cs#L229) (which does not filter out duplicates). In my opinion this is a bad behavior, however I would not change it in this PR as it is already under consideration to be removed.

The merge behavior however calls the [Union](https://github.com/SonarSource/sonar-scanner-msbuild/blob/05499245d82522b5543db07355cc20b10631335a/src/SonarScanner.MSBuild.Tasks/GetAnalyzerSettings.cs#L355) method (which will remove duplicates). And this is the correct behavior, as only those analyzers will be de-duplicated which are exactly the same. Meaning it will not affect the behavior of referencing different versions of the same analyzer.